internal/job/integration/redaction_integration_test.go

```diff
@@ -0,0 +1,164 @@
+package integration
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/buildkite/agent/v3/jobapi"
+	"github.com/buildkite/bintest/v3"
+)
+
+func TestRedactorRedactsAgentToken(t *testing.T) {
+	t.Parallel()
+
+	tester, err := NewExecutorTester(mainCtx)
+	if err != nil {
+		t.Fatalf("setting up executor tester: %v", err)
+	}
+	defer tester.Close()
+
+	tester.ExpectGlobalHook("command").AndCallFunc(func(c *bintest.Call) {
+		fmt.Fprintf(c.Stderr, "The agent token is: %s\n", c.GetEnv("BUILDKITE_AGENT_ACCESS_TOKEN"))
+		c.Exit(0)
+	})
+
+	err = tester.Run(t)
+	if err != nil {
+		t.Fatalf("running executor tester: %v", err)
+	}
+
+	if !strings.Contains(tester.Output, "The agent token is: [REDACTED]") {
+		t.Fatalf("expected agent token to be redacted, but it wasn't. Full output: %s", tester.Output)
+	}
+}
+
+func TestRedactorDoesNotRedactAgentToken_WhenNotInRedactedVars(t *testing.T) {
+	t.Parallel()
+
+	tester, err := NewExecutorTester(mainCtx)
+	if err != nil {
+		t.Fatalf("setting up executor tester: %v", err)
+	}
+	defer tester.Close()
+
+	tester.ExpectGlobalHook("command").AndCallFunc(func(c *bintest.Call) {
+		fmt.Fprintf(c.Stderr, "The agent token is: %s\n", c.GetEnv("BUILDKITE_AGENT_ACCESS_TOKEN"))
+		c.Exit(0)
+	})
+
+	err = tester.Run(t, `BUILDKITE_REDACTED_VARS=""`)
+	if err != nil {
+		t.Fatalf("running executor tester: %v", err)
+	}
+
+	if !strings.Contains(tester.Output, "The agent token is: test-token-please-ignore") {
+		t.Fatalf("expected agent token to be printed in full, but it wasn't. Full output: %s", tester.Output)
+	}
+}
+
+func TestRedactorAdd_RedactsVarsAfterUse(t *testing.T) {
+	// when this test is run inside an agent (ie in CI), we don't want to connect to the job API of the executor running this test
+	// we want to instead connect to the job API of the executor process we're testing
+	t.Setenv("BUILDKITE_AGENT_JOB_API_TOKEN", "")
+	t.Setenv("BUILDKITE_AGENT_JOB_API_SOCKET", "")
```

Comment:
I'm not sure why this needs to be set here and not passed through `tester.Run(t, ...)` or even set as defaults in `NewExecutorTester` (should _any_ tester be connecting to the executor in CI?) Or maybe I'm forgetting some key detail about how the executor works. 

Either way this can be cleaned up later.

Reply:
hmm, i think you're right. i think i misdiagnosed an issue in CI, it seems to run fine without these stanzas